recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache License 2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        try:
+            if self.settings.arch == "armv8" and self.settings.os == "Android":
+                os_name = "Linux"
+            else:
+                os_name = str(self.settings.os)
+
+            archive_ext = "zip" if os_name == "Windows" else "tar.xz"
+            url = f"https://github.com/bytecodealliance/wasmtime/releases/download/v{self.version}/wasmtime-v{self.version}-{self.settings.arch}-{os_name.lower()}-c-api.{archive_ext}"
+            tools.get(url, strip_root=True, destination=self._source_subfolder)
+        except:
+            raise Exception("Binary does not exist for these settings")
+
+    def package(self):
+        include_path = os.path.join(self._source_subfolder, 'include')
+        self.copy('*.h', dst='include', src=include_path)
+        self.copy('*.hh', dst='include', src=include_path)
+        self.copy('*.hpp', dst='include', src=include_path)
+
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
+
+        self.copy('LICENSE', dst='licenses', src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "wasmtime"
+        self.cpp_info.names["cmake_find_multi_package"] = "wasmtime"
+        if self.options.shared:
+            if self.settings.os == "Windows":
+                self.cpp_info.libs = ["wasmtime.dll"]
+            else:
+                self.cpp_info.libs = ["wasmtime"]
+        else:
+            if self.settings.os == "Windows":
+                self.cpp_info.defines= ["/DWASM_API_EXTERN=", "/DWASI_API_EXTERN="]
```

Comment:
conan should handle the rest


```suggestion
                self.cpp_info.defines= ["WASM_API_EXTERN", "WASI_API_EXTERN"]
```

Reply:
Without `/D` conan generates some strange defines and test_package does not work:
```cmake
set(CONAN_DEFINES_WASMTIME "-DWASM_API_EXTERN"
			"-DWASI_API_EXTERN")
set(CONAN_BUILD_MODULES_PATHS_WASMTIME )
# COMPILE_DEFINITIONS are equal to CONAN_DEFINES without -D, for targets
set(CONAN_COMPILE_DEFINITIONS_WASMTIME "WASM_API_EXTERN"
			"WASI_API_EXTERN")
```

Reply:
This works for a lot of other recipes. https://github.com/conan-io/conan-center-index/search?p=2&q=cpp_info+defines

You need to use it in tandem with the other comment https://github.com/conan-io/conan-center-index/pull/6083/files#r663361146


I update the suggestion with a better syntax

Reply:
It is better syntax, but it is not working ...
Please, check it locally in Windows machine